src/app.jsx

```diff
@@ -104,13 +104,14 @@ export default class App extends React.Component {
     // Switch to the next tab if there is one, or the previous, or home.
     let switchTo = 'home';
     if (openJobs[index]) {
-      switchTo = openJobs[index].navID;
+      switchTo = openJobs[index].metadata.navID;
     } else if (openJobs[index - 1]) {
-      switchTo = openJobs[index - 1].navID;
+      switchTo = openJobs[index - 1].metadata.navID;
     }
+    this.switchTabs(switchTo);
     this.setState({
       openJobs: openJobs
-    }, () => this.switchTabs(switchTo));
+    });
```

Comment:
Moved this out of the callback because it doesn't depend on the state being updated. unless i'm missing something important?

Reply:
I agree also. The idea of switching in the callback was probably taken from a similar routine in `openInvestModel` (line 86) where it may be important to wait until the callback so that react has time to re-render with the newly opened model before trying to switch to it. But that's probably not an issue when closing a model.